Runtime/Event/BehaviourEnabledObserver.cs

```diff
@@ -1,53 +1,151 @@
+
+            StopCoroutine(behaviourCheckRoutine);
+            behaviourCheckRoutine = null;
+        }
+
         protected virtual void OnEnable()
         {
-            InvokeRepeating(nameof(Check), MinimumRepeatRate, MinimumRepeatRate);
+            BeginCheck();
         }
 
         protected virtual void OnDisable()
         {
-            CancelInvoke(nameof(Check));
+            EndCheck();
         }
 
         /// <summary>
-        /// Checks whether all <see cref="behaviours"/> are <see cref="Behaviour.isActiveAndEnabled"/> and emits <see cref="ActiveAndEnabled"/> if they are.
+        /// Checks to see if the <see cref="Behaviours"/> specified have been enabled in the scene.
         /// </summary>
-        protected virtual void Check()
+        /// <returns>An Enumerator to manage the running of the Coroutine.</returns>
+        protected virtual IEnumerator Check()
         {
-            if (!behaviours.TrueForAll(behaviour => behaviour.isActiveAndEnabled))
+            checkDelayYieldInstruction = new WaitForSeconds(CheckDelay);
+            timeUntilCheckIsCancelled = Time.time + MaximumRunTime;
+            while (Time.time < timeUntilCheckIsCancelled)
             {
-                return;
+                if (AreBehavioursEnabled())
+                {
+                    break;
+                }
+                yield return checkDelayYieldInstruction;
+            }
+            behaviourCheckRoutine = null;
+        }
+
+        /// <summary>
+        /// Checks whether all <see cref="Behaviours"/> are <see cref="Behaviour.isActiveAndEnabled"/> and emits <see cref="ActiveAndEnabled"/> if they are.
+        /// </summary>
+        /// <returns>Whether all <see cref="Behaviours"/> are active and enabled.</returns>
+        protected virtual bool AreBehavioursEnabled()
+        {
+            if (Behaviours == null || Behaviours.NonSubscribableElements.Count == 0)
+            {
+                return false;
+            }
+
+            foreach (Behaviour behaviour in Behaviours.NonSubscribableElements)
+            {
+                if (!behaviour.isActiveAndEnabled)
+                {
+                    return false;
+                }
             }
 
-            CancelInvoke(nameof(Check));
             ActiveAndEnabled?.Invoke();
+            return true;
+        }
+
+        /// <summary>
+        /// Called after <see cref="MaximumRunTime"/> has been changed.
+        /// </summary>
+        [CalledAfterChangeOf(nameof(MaximumRunTime))]
+        protected virtual void OnAfterMaximumRunTimeChange()
+        {
+            float remainingRunTime = timeUntilCheckIsCancelled - Time.time;
+            timeUntilCheckIsCancelled = MaximumRunTime - remainingRunTime;
+        }
+
+        /// <summary>
+        /// Called after <see cref="CheckDelay"/> has been changed.
+        /// </summary>
+        [CalledAfterChangeOf(nameof(CheckDelay))]
+        protected virtual void OnAfterCheckDelayChange()
```

Comment:
Move this up so it's before `OnAfterMaximumRunTimeChange` to match the order of the properties' definition.

Reply:
👍